### PR TITLE
General info controller and 2 more 

### PR DESCRIPTION
**Pull request by tolyara**

### CODE REVIEW

File: src/main/java/com/Kh033Java/travelbook/controller/map_controller/MapController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	
+	@Autowired
+	CountryService countryService;
+
+	@RequestMapping(value = "users/{login}/map", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+	public List<Country> getCountryCities(@PathVariable String login) {
+		return countryService.getVisitedCountries(login);
+	}
+
+}
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/controller/map_controller/MapController.java]: add also planned trips

File: src/main/java/com/Kh033Java/travelbook/repository/CountryRepository.java [hunk 2]
```diff
@@ -12,8 +12,10 @@
 	
     List<Country> findAll();
     
-//    @Query("Match (c:Country)<-[v:VISITED]-(p:User) where c.name={countryName} return c,p")
     @Query("MATCH (n) WHERE n.name={countryName} RETURN n")
     Country getCountryByName(@Param("countryName") String countryName);
+
+    @Query("match (c:Country)<-[v:VISITED]-(u:User)-[cr:CREATED]->(n:Note) where u.login={userLogin} and n.isPublic=true return c")
+	List<Country> getCountriesWhereNotesPublic(@Param(value = "userLogin") String userLogin);
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/repository/CountryRepository.java]: change `[cr:CREATED]` to `[cr:CREATED_NOTE]`. look at User entity for naming relationships

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
+package com.Kh033Java.travelbook.controller.general_info_controller;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.runner.ReactiveWebApplicationContextRunner;
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java]: delete

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/WeatherProvider.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.net.*;
+import java.net.URL;
+import java.text.DecimalFormat;
+import java.util.*;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/WeatherProvider.java]: delete

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+	CountryService countryService;
+
+	@Autowired
+	CityService cityService;
+	
+	@Autowired
+	DescriptionService descriptioService;
+
+	@RequestMapping(value = "/country/{location}/description/weather", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+	public Weather getWeather(@PathVariable String location) {
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java]: what is location?

File: src/main/java/com/Kh033Java/travelbook/controller/map_controller/MapController.java (hunk 1 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/controller/map_controller/MapController.java]: done

File: src/main/java/com/Kh033Java/travelbook/repository/CountryRepository.java (hunk 2 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/repository/CountryRepository.java]: done

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java (hunk 3 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java]: annotation Autowired needed, unnecessary import deleted

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/WeatherProvider.java (hunk 4 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/WeatherProvider.java]: done

File: src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java (hunk 5 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/controller/general_info_controller/GeneralInfoController.java]: Changed on "countryName"

File: src/main/java/com/Kh033Java/travelbook/controller/GeneralInfoController.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	
+	@Autowired
+	DescriptionService descriptionService;
+
+	@RequestMapping(value = "/country/{countryName}/description", method = RequestMethod.GET, produces = MediaType.APPLICATION_JSON_VALUE)
+	public String getGeneralInfo(@PathVariable String countryName) {
+		final Country country = countryService.getByName(countryName);
+		final Description description = descriptionService.getDesccriptionByCountryName(country.getName());
+		final Weather weather = getWeatherInCapital(description);
+		final CountryGeneralInfoDTO countryGeneralInfoDTO = new CountryGeneralInfoDTO(country.getName(), description, weather);
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/controller/GeneralInfoController.java]: why do we need get country from db and then just call `country.getName()` if we could get country name from path variable `{countryName}`?

File: src/main/java/com/Kh033Java/travelbook/repository/DescriptionRepository.java [hunk 7]
```diff
@@ -1,9 +1,19 @@
...
+import org.springframework.data.neo4j.annotation.Query;
+import org.springframework.data.neo4j.repository.Neo4jRepository;
+import org.springframework.data.repository.query.Param;
+
+import com.Kh033Java.travelbook.entity.Country;
+import com.Kh033Java.travelbook.entity.Description;
+
+import java.util.List;
+
+public interface DescriptionRepository extends Neo4jRepository<DescriptionRepository, Long> {
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/repository/DescriptionRepository.java]: change from `<DescriptionRepository, Long>` to `<Description, Long>`

File: src/main/java/com/Kh033Java/travelbook/service/PlanService.java [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+import com.Kh033Java.travelbook.entity.Plan;
+import com.Kh033Java.travelbook.repository.PlanRepository;
+
+@Service
+public class PlanService {
+	
+    @Autowired
+    PlanRepository planRepository;
+
+	public List<Plan> getPlannedTrips() {
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/service/PlanService.java]: unused method

File: src/main/java/com/Kh033Java/travelbook/util/WeatherProvider.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+			URL url = new URL(URL_STRING);
+			URLConnection connection = url.openConnection();
+			BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+			String line;
+			while ((line = reader.readLine()) != null) {
+				result.append(line);
+			}
+			reader.close();
+		} catch (IOException e) {
+			e.printStackTrace();
```
Comment (vanilmalik) [inline src/main/java/com/Kh033Java/travelbook/util/WeatherProvider.java]: change to logger instead printStackTrace

File: src/main/java/com/Kh033Java/travelbook/controller/GeneralInfoController.java (hunk 6 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/controller/GeneralInfoController.java]: fixed

File: src/main/java/com/Kh033Java/travelbook/repository/DescriptionRepository.java (hunk 7 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/repository/DescriptionRepository.java]: done

File: src/main/java/com/Kh033Java/travelbook/service/PlanService.java (hunk 8 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/service/PlanService.java]: all class is unused now and have been deleted

File: src/main/java/com/Kh033Java/travelbook/util/WeatherProvider.java (hunk 9 above)
Comment (tolyara) [reply src/main/java/com/Kh033Java/travelbook/util/WeatherProvider.java]: done

**Pull request merged**